pkg/cloud/gce.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 Elotl Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+const (
+	metadataAliasCIDR = "instance/network-interfaces/0/ip-aliases/0"
+	metadataIPv4      = "instance/network-interfaces/0/ip"
+)
+
+type itzoTransport struct {
+	userAgent string
+	base      http.RoundTripper
+}
+
+func (t itzoTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", t.userAgent)
+	return t.base.RoundTrip(req)
+}
+
+type GceCloudInfo struct {
+	metadata *metadata.Client
+}
+
+func NewGceCloudInfo() (CloudInfo, error) {
+	c := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: itzoTransport{
+			userAgent: "elotl/itzo",
+			base:      http.DefaultTransport,
+		},
+	}
+	metadataClient := metadata.NewClient(c)
+	return &GceCloudInfo{
+		metadata: metadataClient,
+	}, nil
+}
+
+func (g *GceCloudInfo) GetMainIPv4Address() (string, error) {
+	if !metadata.OnGCE() {
+		return "", fmt.Errorf("instance is not running inside GCE, could not retrieve Node IP")
+	}
+
+	addr, err := g.metadata.Get(metadataIPv4)
```

Comment:
you can just `return g.metadata.Get(metadataIPv4)` instead of checking the error.